src/blib2to3/Grammar.txt

```diff
@@ -247,5 +247,5 @@ subject_expr: (namedexpr_test|star_expr) (',' (namedexpr_test|star_expr))* [',']
 # cases
 case_block: "case" patterns [guard] ':' suite
 guard: 'if' namedexpr_test
-patterns: pattern ['as' pattern]
-pattern: (expr|star_expr) (',' (expr|star_expr))* [',']
+patterns: pattern (',' pattern)* [',']
```

Comment:
maybe i'm missing something, but this seems to allow a trailing comma?

it's legal syntax:

```python
>>> x = 1
>>> match x:
...     case 1,:
...         print('yep')
... 
```

…but the trailing `,` is part of the pattern, turning it into a tuple, as this example shows:

```python
>>> x = (1,)
>>> match x:
...     case 1,:
...         print('yep')
... 
yep
```

Reply:
I am unsure what do you mean by it is turning into a tuple? The grammar has no semantical meaning, it only defines what is accepted as is or not. Would you mind giving a bit more explanation? The trailing comma is a valid syntax AFAIK.

Reply:
nevermind, i though the possibility of a trailing might've been included already in the `expr` on the line below already, but that seems not the case.